HIP/hip-0000-cancun-2.md

```diff
@@ -0,0 +1,155 @@
+---
+hip: <HIP number (this is determined by the HIP editor)>
+title: Add EVM compatibility for non-supported Cancun blob features.
+author: Danno Ferrin (@shemnon)
+working-group: // TBD
+type: Standards Track
+category: Core
+needs-council-approval: Yes
+status: Draft
+created: // TBD
+discussions-to: // TBD
+updated: <comma separated list of dates>
+requires: <HIP number(s)>
+replaces: <HIP number(s)>
+superseded-by: <HIP number(s)>
+---
+
+## Abstract
+
+Add compatibility for Ethereum Mainnet Cancun features relating to Hedera's
+non-support of EIP-4844 blobs.
+
+## Motivation
+
+In order to maintain maximum compatibility with Ethereum Mainnet there will need
+to be support for some Blob opcodes in the EVM that behave appropriately in the
+absence of Blobs support in Hedera. At the same time, space needs to be
+preserved in the event Hedera provides similar Blob functionality that needs to
+be mapped to the [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844) facilities.
+
+## Rationale
+
+In order to both preserve compatibility and future design space we need to act
+as if blobs are not being added to Hedera. This allows existing contracts that
+may depend on blob behavior to continue to function in the absence of blobs.
+
+First we will prevent blobs from entering the system. There are complex EL/CL
+interactions in Ethereum Mainnet, but as far as Hedera's consensus is similar to
+Ethereum Mainnet's, the main entry point is a "Type 3" transaction that allows
+blobs to be attached. Furthermore, Type 3 transactions require a blob to be
+attached. Prohibiting Type 3 transactions will thus be sufficient to keep blobs
+form entering the EVM's realm of concern and similarly will not prohibit other
+interactions that are desirable to Hedera.
+
+Second, we need to support the new opcodes but ensure they operate as they would
+in the absence of any Blobs.
+
+## User stories
+
+* As a smart contract deployer, I want my smart contract that expects blob
+  support to not fail in unexpected ways.
+* As a smart contract user, I will not be able to successfully use contracts
+  that depend on the presence of blobs.
+* As a future Hedera Core developer, I want the option to provide blob like
+  behavior using EIP-4844 semantics and existing opcodes.
+
+## Specification
+
+### EVM Implementation
+
+Two opcodes need to be supported. The `VERSIONEDHASH` opcode defined
+in EIP-4844 and the `BLOBBASEFEE`
+operation in [EIP-7516](https://eips.ethereum.org/EIPS/eip-7516). These opcodes
+should behave in the same way as though there are no blobs and that there never
+have been blobs.
+
+For `VERSIONEDHASH` this will result in returning all zeros whenever called, as
```

Comment:
DO these 2 changes require Besu CLient code overrides?
If so we should note it so it's not confused for default or configureable Besu logic.

Reply:
Added a clarifying paragraph at the end.  No changes to besu code.